packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -2170,6 +2171,20 @@ function commitMutationEffectsOnFiber(
             }
           }
         }
+      } else {
+        if (enableEagerAlternateStateNodeCleanup) {
+          if (supportsPersistence) {
```

Comment:
I think what you have here is fine, but I think this is what @rickhanlonii meant was this.
```suggestion
      } else if (supportsPersistence) {
        if (enableEagerAlternateStateNodeCleanup) {
```